contracts/staking/StakeManager.sol

```diff
@@ -63,37 +70,46 @@ contract StakeManager is Validator, IStakeManager, RootChainable, Lockable {
   //Mapping for epoch to totalStake for that epoch
   mapping (uint256 => State) public validatorState;
 
-  constructor () ERC721Full("Matic Validator", "MV") public {}
+  constructor (address _registry) ERC721Full("Matic Validator", "MV") public {
+    registry = _registry;
+  }
 
   // only staker
   modifier onlyStaker(uint256 validatorId) {
     require(ownerOf(validatorId) == msg.sender);
     _;
   }
 
-  function stake(uint256 amount, address signer) public {
-    stakeFor(msg.sender, amount, signer);
+  function stake(uint256 amount, address signer, bool isContract) external {
+    stakeFor(msg.sender, amount, signer, isContract);
   }
 
-  function stakeFor(address user, uint256 amount, address signer) public onlyWhenUnlocked {
+  function stakeFor(address user, uint256 amount, address signer, bool isContract) public onlyWhenUnlocked {
     require(currentValidatorSetSize() < validatorThreshold);
     require(balanceOf(user) == 0, "No second time staking");
     require(amount > MIN_DEPOSIT_SIZE);
     require(signerToValidator[signer] == 0);
```

Comment:
Do we really need to check this if we are checking `balanceOf(user)`?

Reply:
that's kinda tricky hack to make sure that we have unique validators(user address).